app/models/checkbox_answers.rb

```
@@ -2,7 +2,9 @@ class CheckboxAnswers < ActiveRecord::Base
   self.implicit_order_column = "created_at"
   belongs_to :step
 
-  validates :response, presence: true
+  validates :response,
+    presence: true,
+    unless: proc { |answer| answer.step && answer.step.skippable? && answer.skipped }
```

Is the `answer.step` needed here? We can't have an answer without a step, or if we do then it probably should throw an error.

---

This was added because of the test suite. For the Shoulda helpers to continue working they would be built without an associated step (the step would be added on `create`). I've remove the Shoulda matchers now and gone manual as you're right, we should never have an answer without a step, in real life.